web/modules/custom/usergreeting/src/Plugin/Block/UsergreetingBlock.php

```diff
@@ -0,0 +1,68 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\usergreeting\Plugin\Block;
+
+use Drupal\Core\Block\BlockBase;
+use Drupal\Core\Plugin\ContainerFactoryPluginInterface;
+use Drupal\usergreeting\GreetingTime;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Provides a 'Usergreeting' Block.
+ *
+ * @Block(
+ *   id = "usergreeting_block",
+ *   admin_label = @Translation("Usergreeting"),
+ *   category = @Translation("Usergreeting"),
+ * )
+ */
+class UsergreetingBlock extends BlockBase implements ContainerFactoryPluginInterface {
+
+  /**
+   * The greeting message.
+   *
+   * @var \Drupal\usergreeting\GreetingTime
+   */
+  protected $greeting;
+
+  /**
+   * Construct a UsergreetingBlock object.
+   *
+   * @param array $configuration
+   *   A configuration array containing information about the plugin instance.
+   * @param string $plugin_id
+   *   The plugin ID for the plugin instance.
+   * @param mixed $plugin_definition
+   *   The plugin implementation definition.
+   * @param \Drupal\usergreeting\GreetingTime $greeting
+   *   The greeting message.
+   */
+  public function __construct(array $configuration, string $plugin_id, $plugin_definition, GreetingTime $greeting) {
+    parent::__construct($configuration, $plugin_id, $plugin_definition);
+    $this->greeting = $greeting;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container, array $configuration, $plugin_id, $plugin_definition) {
+    return new static($configuration, $plugin_id, $plugin_definition,
+      $container->get('usergreeting.greeting')
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function build() {
+    $build = [];
+    $greetuser = $this->greeting->greetingMessage();
+
+    $build['#markup'] = '<p>' . $greetuser . '</p>';
```

Comment:
Using `#markup` in render arrays is unsafe as it might not be filtered according to [the Markup object docs](https://api.drupal.org/api/drupal/core%21lib%21Drupal%21Core%21Render%21Markup.php/class/Markup/8.2.x). Let's say the username somehow contains XSS code which is then rendered on the page, resulting in the script executed. For your case it's preferred to use a different element for rendering, e.g.:

https://api.drupal.org/api/drupal/core%21lib%21Drupal%21Core%21Render%21Element%21HtmlTag.php/class/HtmlTag/8.2.x